### ISSUE TITLE
Newman does not replace variables from environment file

### ISSUE THREAD

**Issue by fstorz** [postmanlabs/newman#1336]
1. Newman Version (can be found via `newman -v`):
```
newman -v
3.9.1
```
 2. OS details (type, version, and architecture): Windows
 3. Are you using Newman as a library, or via the CLI? CLI
 3. Did you encounter this recently, or has this bug always been there: recently
 4. Expected behaviour:
Newman replaces all environment variables in the postman collection
 5. Command / script used to run Newman:
`newman run -e TEST.postman_environment.json --insecure --no-color Test.postman_collection.json`
 6. Sample collection, and auxiliary files (minus the sensitive details):
Test.postman_collection.json
```
{
	"info": {
		"name": "Test",
		"_postman_id": "48efccfb-67d6-ade1-8a98-6bc6b98e0c37",
		"description": "",
		"schema": "https://schema.getpostman.com/json/collection/v2.1.0/collection.json"
	},
	"item": [
		{
			"name": "Test",
			"request": {
				"method": "GET",
				"header": [],
				"body": {},
				"url": {
					"raw": "{{host}}?q=test",
					"host": [
						"{{host}}"
					],
					"query": [
						{
							"key": "q",
							"value": "test",
							"equals": true
						}
					]
				},
				"description": ""
			},
			"response": []
		}
	]
}
```
TEST.postman_environment.json
```
{
  "id": "4d8a1864-96ec-d01a-2663-440e70ee9d15",
  "name": "TEST",
  "values": [
    {
      "key": "host",
      "value": "http://google.de",
      "description": "",
      "type": "text",
      "enabled": true
    }
  ],
  "timestamp": 1512988941001,
  "_postman_variable_scope": "environment",
  "_postman_exported_at": "2017-12-11T10:43:25.859Z",
  "_postman_exported_using": "Postman/5.5.0"
}
```

The actual result looks as follows
```
> newman run -e TEST.postman_environment.json --insecure --no-color Test.postman_collection.json
newman

Test

→ Test
  GET {{host}}?q=test
┌─────────────────────────┬──────────┬──────────┐
│                         │ executed │   failed │
├─────────────────────────┼──────────┼──────────┤
│              iterations │        1 │        0 │
├─────────────────────────┼──────────┼──────────┤
│                requests │        1 │        1 │
├─────────────────────────┼──────────┼──────────┤
│            test-scripts │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│      prerequest-scripts │        0 │        0 │
├─────────────────────────┼──────────┼──────────┤
│              assertions │        0 │        0 │
├─────────────────────────┴──────────┴──────────┤
│ total run duration: 67ms                      │
├───────────────────────────────────────────────┤
│ total data received: 0B (approx)              │
├───────────────────────────────────────────────┤
│ average response time: 0ms                    │
└───────────────────────────────────────────────┘

  #  failure                          detail

 1.  Error                            Invalid URI "http:///%7B%7Bhost%7D%7D?q=test"
                                      at request
                                      inside "Test"
```

**Comment by fstorz** [comment]
Problem is solved when I adjust the command as follows
`newman run Test.postman_collection.json -e TEST.postman_environment.json --insecure --no-color`

But still, the old way should also work, or you may point to this breaking change in the docu somewhere

**Comment by kunagpal** [comment]
@deepakpathania

**Comment by deepakpathania** [comment]
Hey @fstorz, thanks for reaching out. 

`newman run` actually expects a collection path to be followed immediately, as specified [here](https://github.com/postmanlabs/newman#newman-run-collection-file-source-options). The correct specification is `newman run <collection-file-source> [options]` so your options like `-e` should be followed after the collection file source is specified as you did in your second attempt. 

I believe the documentation is explicit enough to specify this but if you feel it is a little confusing, please point me to the part which lead to the confusion and I would be happy to make the appropriate changes.

**Comment by ClayBenson94** [comment]
@deepakpathania has this always been the case? I'm on version 3.9.1 and I seem to be able to do both orders (using the `-g ./globals.json` flag before the collection, or after.

**Comment by deepakpathania** [comment]
@ClayBenson94 The ordering works in your case because of a fallback that we have that checks for the collection path in the CLI args which was intended for cases when newman was bein required as a module. The order that I suggested above is the recommended way though. You can verify this by running something like: 
`newman run -n 10 examples/sample-collection.json` 

The collection would run but only once rather than 4 times like the intended behavior, since collection path would be recognized through the args fallback but the options specified above wouldn't take effect.

**Comment by fstorz** [comment]
@deepakpathania my problem was that i used my config with the old version of newman for a while. And now when I wanted to test the new version, it wasn't clear to me that you have changed the cli command order and I din't also thought about that this could be the problem. 
Because there isn't any warning, that the parameters are not recognized/used, it might be good to show something like 'the parameter -e is not recognized'.

**Comment by deepakpathania** [comment]
@fstorz Yeah, makes sense. We will try to add some explicit warning for options that were supplied but not used/ recognized. We'll also try and see if the run command usage can be made backward compatible so that the older configs can still work with it.

**Comment by deepakpathania** [comment]
Hey @fstorz, we are working on fixing this issue, you can track progress in #1343.
Thanks for your patience.

**Comment by deepakpathania** [comment]
Hey @fstorz, After a lot of discussions, we have decided not to support this since we would like to stick to the `newman run <collection-file-source> [options]` as the only supported specification moving forward. I would recommend you make the changes accordingly. 
Thanks for pointing this out though.

**Comment by dilp** [comment]
> Problem is solved when I adjust the command as follows
> `newman run Test.postman_collection.json -e TEST.postman_environment.json --insecure --no-color`
> 
> But still, the old way should also work, or you may point to this breaking change in the docu somewhere

This is not working for me...

**Comment by nunez-giovanni** [comment]
@dilp - Did you ever figure out a solution? 
I noticed `newman run <your test collection> -e <your test environment> --insecure --no-color` did not work either. 

If I manually change the environment variable to the expected value ( in my test collection ) it works :-/ so not sure why the test environment isn't being read correctly. 

I know the test environment json is being read, as there is another ENV value in there, and that is getting passed correctly. 

exported the ENV from postman to make sure I was not generating an invalid JSON file.

**Comment by BugliL** [comment]
I am having the same problem

**Comment by vijayarondla-mosaic** [comment]
I a having the same issue
